Use newer firstIndex(of:) in Swift 5+

efirestone opened the pull request.

gfontenot commented on Sources/String+Template.swift:
```diff
@@ -176,15 +176,25 @@ extension String {
-        var formatStringIndex = formatted.index(of: template.placeholderToken)
+        var formatStringIndex: String.Index?
+        #if swift(>=5)
+            formatStringIndex = formatted.firstIndex(of: template.placeholderToken)
```
Thoughts on polyfilling this instead? Basically define `firstIndex(of:)` for Swift < 5.0 in a different file so that we don't litter this code with version-specific changes?

efirestone replied:
Turns out `firstIndex(of:)` is available in Swift 4 (and maybe before), so no need to polyfill or add conditionals.

efirestone committed:
Use newer String.firstIndex(of:)

`firstIndex(of:)` is available in Swift 4, and the older `index(of:)` method is now officially deprecated.

efirestone commented:
I don't have merge rights, so please merge if it looks good.
